bugfix: Fix topic editing if topic unchanged

neiljp opened the pull request:
Recent work in resolved/unresolved topics has caused a change in the server (not yet in a release), where it more strictly enforces that if a topic is provided but unchanged in stream message editing/updating, then the propagate mode must be set to change_one - otherwise the edit does not proceed.

This PR works around this and has scope to be expanded towards resolving part or all of #1014.

The need for this was discussed in #**design>closing/solving a topic (#11154)**.

I've not added tests for this change in behavior (yet, at least) - we don't seem to have tests for the existing behavior.

preetmishra commented on zulipterminal/ui_tools/boxes.py:
```diff
@@ -539,10 +539,15 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
                     topic = self.title_write_box.edit_text
 
                 if self.msg_edit_state is not None:
+                    if topic == self.msg_edit_state[1]:  # FIXME need strip?
```
For a topic change from "Test" to "Test ", it says invalid prop mode. I think we need to strip here.

preetmishra reviewed (commented):
@neiljp Thanks for pushing a patch! It seems to work well locally. :+1: 

Re the state data structure, fetching individual items from a `tuple` (using indices) doesn't tell much about what is being extracted out of the tuple. What do you think about a `dict` instead?

I have left a minor comment regarding a FIXME comment.

neiljp commented:
Just returned to this after my time away as it blocks our use on czo with simple editing.

@preetmishra Just pushed an updated version addressing your points:
* using a namedtuple instead of a simple tuple (dicts are OK but are more verbose and less constrained unless one uses typeddict, and that's only enforced via mypy)
* uses the trimmed topic for comparison purposes (with notes to explain why - I checked the server code)

The namedtuple change actually found a minor test change that was missing too, so thanks for the suggestion to avoid indices (even if internal to the WriteBox)

preetmishra commented on zulipterminal/ui_tools/boxes.py:
```diff
@@ -543,9 +544,16 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
                         message_id=self.msg_edit_state.message_id,
-                        topic=topic,
+                        topic=topic,  # NOTE: Send untrimmed topic always for now
                         propagate_mode=self.edit_mode_button.mode,
```
This should be `propagate_mode=propagate_mode`.

preetmishra reviewed (changes requested):
@neiljp Thanks for pushing the updates! :+1:

The PR doesn't work currently but that's due to a small mishap (see my in-line comment). Other than that, the changes look good to me.

NamedTuple is even better with strict checking. Thanks for moving ahead with it. I second your decision regarding only using a trimmed topic name for equality and sending the topic as is until we have a guideline from the server.

neiljp commented:
@preetmishra Thanks for the reviews! Merging this now so I can get back to editing messages! :tada:
